### PR TITLE
Add more details to labels model in 5.0 guide

### PR DESCRIPTION
**Pull request by zli82016**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
docs: added more details to labels model in 5.0 guide
```

### CODE REVIEW

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 1]
```diff
@@ -96,28 +96,99 @@ Not all of Google cloud resources support labels and annotations. Please check t
...
+
+**Note:** `ignore_changes` can be applied to `labels` field to ignore the changes of the user defined labels. It is not recommended to apply `ignore_changes` to `terraform_labels` or `effective_labels`, as it will make the new labels model not work as expected.
 
-After upgrading to `5.0.0`, and then running `terraform refresh` or `terraform apply`, these three fields should show in the state file of the resources with a self-applying `labels` field.
+The following changes will be observed after upgrading to `5.0.0`.
+* After upgrading to `5.0.0`, and then running `terraform import`, these three fields should show in the state file of the resources with empty `labels` and empty `terraform_labels`. `effective_labels` will have all of labels present on the resource in GCP.
+* After upgrading to `5.0.0`, and then running `terraform refresh`, these three fields should show in the state file of the resources with a self-applying `labels` field and empty `terraform_labels`. `effective_labels` will have all of labels present on the resource in GCP.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:119]: "with a self-applying `labels` field" what does this mean?

I think we expect to see the current set of labels in state preserved as the value for `labels`, right?

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 2]
```diff
@@ -96,28 +96,99 @@ Not all of Google cloud resources support labels and annotations. Please check t
...
+* After upgrading to `5.0.0`, and then running `terraform plan` or `terraform apply`, the plan shows that a new block `terraform_labels` will be added to the state. After running `terraform apply`, these three fields should show in the state file of the resources with a self-applying `labels` field. In the state file, `terraform_labels` will have the merged labels and `effective_labels` will have all of labels present on the resource in GCP.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:120]:
```suggestion
* After upgrading to `5.0.0`, and then running `terraform plan` or `terraform apply`, the plan will show an updated adding your current labels to `terraform_labels`. After running `terraform apply`, these three fields should show in the state file of the resources with a self-applying `labels` field. In the state file, `terraform_labels` will have the combination of `labels` and your provider-default labels and `effective_labels` will have all of labels present on the resource in GCP.
```

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 3]
```diff
@@ -96,28 +96,99 @@ Not all of Google cloud resources support labels and annotations. Please check t
...
+* In the resource `google_cloud_run_domain_mapping`, the system labels `cloud.googleapis.com/location` and `run.googleapis.com/overrideAt` are removed from `labels` inside `metadata` field in the state file. If any of these label keys are in the configration, after upgrading to `5.0.0`, the plan shows that these keys will be added.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:121]:
```suggestion
* In the resource `google_cloud_run_domain_mapping`, the system labels `cloud.googleapis.com/location` and `run.googleapis.com/overrideAt` are removed from `labels` inside `metadata` field in the state file. If any of these label keys are in the configuration, after upgrading to `5.0.0`, the plan shows that these keys will be added. You can safely accept this change.
```

Comment (rileykarson) [reply mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:121]: (same for others below)

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 4]
```diff
@@ -96,28 +96,99 @@ Not all of Google cloud resources support labels and annotations. Please check t
...
+* In the resource `google_notebooks_instance`, the system labels with the prefix `goog-caip-notebook` are removed from the `labels` field in the state file. If any of these system label keys are in the configration, after upgrading to `5.0.0`, the plan shows that these keys will be added.
+* In the resource `google_storage_bucket`, the system labels with the prefix `goog-dataplex` are removed from the `labels` field in the state file. If any of these system label keys are in the configration, after upgrading to `5.0.0`, the plan shows that these keys will be added.
+
+#### Data source labels
+
+Since the resource has used the new labels model, the corresponding data source now has three label-related fields as well, `labels`, `effective_labels` and `terraform_labels`. All of these three fields will have all of labels present on the resource in GCP, including the labels configured through Terraform, other clients and services.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:131]:
```suggestion
For most resource-based datasources, all three of `labels`, `effective_labels` and `terraform_labels` will now be present. All of these three fields will have all of labels present on the resource in GCP including the labels configured through Terraform, the system, and other clients, equivalent to `effective_labels` on the resource.
```

**Review by rileykarson — COMMENTED** [review]
Note: I commented on `labels`, but same transformation where relevant for `annotations`.

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown (hunk 1 above)
Comment (zli82016) [reply mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown]: that is right. Changed.

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 5]
```diff
@@ -96,27 +96,98 @@ Not all of Google cloud resources support labels and annotations. Please check t
...
 * The output-only `terraform_labels` will merge the labels defined by the users on the resource through Terraform and the default labels configured on the provider. If the same label exists on both the resource labels and provider default labels, the label on the resource will override the provider label.
+* The output-only `effective_labels` will list all of labels present on the resource in GCP, including the labels configured through Terraform, the system, and other clients.
+
+**Note:** `ignore_changes` can be applied to `labels` field to ignore the changes of the user defined labels. It is not recommended to apply `ignore_changes` to `terraform_labels` or `effective_labels`, as it may unintuitively affect the final API call.
+
+The following changes will be observed after upgrading to `5.0.0`.
+* Running `terraform import` on Google provider `5.0.0` or later, these three fields will show in the state file with an empty `labels` and `terraform_labels` value. `effective_labels` will have all of labels present on the resource in GCP. You can update the resource to bring labels back under management.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:118]:
```suggestion
* Running `terraform import` on Google provider `5.0.0` or later, these three fields will show in the state file with an empty `labels` and `terraform_labels` value. `effective_labels` will have all of labels present on the resource in GCP. You can update the resource to bring labels defined in your configuration under management by Terraform.
```

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 6]
```diff
@@ -96,27 +96,98 @@ Not all of Google cloud resources support labels and annotations. Please check t
...
+* Running `terraform refresh` on Google provider `5.0.0` or later, these three fields will show in the state file. `labels` field will have your current labels, `terraform_labels` will be empty, and `effective_labels` will have all of labels present on the resource in GCP.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:119]:
```suggestion
* Running `terraform refresh` on Google provider `5.0.0` or later with an existing pre-`5.0.0` resource before an `apply`, these three fields will show in the state file. `labels` field will have your current labels, `terraform_labels` will be empty, and `effective_labels` will have all of labels present on the resource in GCP.
```

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 7]
```diff
@@ -96,27 +96,98 @@ Not all of Google cloud resources support labels and annotations. Please check t
...
+* Running `terraform plan` or `terraform apply` on Google provider `5.0.0` or later, the plan will show an updated adding your current labels to `terraform_labels`. After running `terraform apply`, these three fields will show in the state file. `labels` will have your current labels, `terraform_labels` will have the combination of `labels` and your provider-default labels, and `effective_labels` will have all of labels present on the resource in GCP.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:120]:
```suggestion
* Running `terraform plan` or `terraform apply` on Google provider `5.0.0` or later with an existing pre-`5.0.0` resource before an `apply`, the plan will show an updated adding your current labels to `terraform_labels`. After running `terraform apply`, these three fields will show in the state file. `labels` will have your current labels, `terraform_labels` will have the combination of `labels` and your provider-default labels, and `effective_labels` will have all of labels present on the resource in GCP.
```

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 8]
```diff
@@ -96,27 +96,98 @@ Not all of Google cloud resources support labels and annotations. Please check t
...
+* In the resource `google_cloud_run_domain_mapping`, the system labels `cloud.googleapis.com/location` and `run.googleapis.com/overrideAt` are removed from `labels` inside `metadata` field in the state file. If any of these label keys are in the configuration, after upgrading to `5.0.0`, the plan shows that these keys will be added. You can safely accept this change.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:121]:
```suggestion
* In the resource `google_cloud_run_domain_mapping`, the system labels `cloud.googleapis.com/location` and `run.googleapis.com/overrideAt` will be removed from `labels` inside `metadata` field in the state file as part of a one-time resource schema upgrade. If any of these label keys are in the configuration after upgrading to `5.0.0`, the plan will show these keys will be added. You can safely accept this change, and Terraform will begin to manage them again.
```

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 9]
```diff
@@ -96,27 +96,98 @@ Not all of Google cloud resources support labels and annotations. Please check t
...
+  name   = "tf-test-dproc-test-1"
+  region = "us-central1"
+
+  labels = {
+    key1 = "value1"
+    key2 = "value2"
+  }
+}
+```
+After the configration is applied, Terraform is managing `key1` and `key2` in the `labels` field. `terraform_labels` field has label `default_key`, `key1` and `key2`. `effective_labels` has label `default_key`, `key1`, `key2` and system labels.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:171]:
```suggestion
After the configuration is applied, Terraform is managing `key1` and `key2` in the `labels` field. `terraform_labels` field has label `default_key`, `key1` and `key2`. `effective_labels` has label `default_key`, `key1`, `key2` and system labels.
```

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 10]
```diff
@@ -96,27 +96,98 @@ Not all of Google cloud resources support labels and annotations. Please check t
...
+resource "google_dataproc_cluster" "with_labels" {
+  name   = "tf-test-dproc-test-1"
+  region = "us-central1"
+
+  labels = {
+    key1 = "value1"
+  }
+}
+```
+After the configration is applied, Terraform stops managing `key2` and is managing `key1` in the `labels` field. `terraform_labels` field has label `default_key` and `key1`. `effective_labels` has label `default_key`, `key1` and system labels.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:190]:
```suggestion
After the configuration is applied, Terraform stops managing `key2` and is managing `key1` in the `labels` field. `terraform_labels` field has label `default_key` and `key1`. `effective_labels` has label `default_key`, `key1` and system labels.
```

**Review by rileykarson — APPROVED** [review]
Made some suggestions- if you can apply them in similar spots (i.e. labels suggestions against annotations entries, and the stateupgrader suggestion to the entire list of cases, LGTM

**Comment by zli82016** [comment]
Thanks so much for the comments.

**Pull request merged**